#! /bin/sh -e

# DP: Support -ffast-math on hurd-i386.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

--- gcc/config.gcc	2007-11-16 15:27:19.000000000 +0100
+++ gcc/config.gcc.new	2007-11-16 15:30:25.000000000 +0100
@@ -1109,6 +1109,7 @@
 	tmake_file="${tmake_file} i386/t-linux64 i386/t-crtstuff i386/t-crtfm t-dfprules"
 	;;
 i[34567]86-*-gnu*)
+	tmake_file="${tmake_file} i386/t-crtfm"
 	;;
 i[34567]86-pc-msdosdjgpp*)
 	xm_file=i386/xm-djgpp.h
